src/backend/distributed/transaction/backend_data.c

```diff
@@ -845,7 +845,7 @@ ActiveDistributedTransactionNumbers(void)
 	int curBackend = 0;
 
 	/* build list of starting procs */
-	for (curBackend = 0; curBackend < MaxBackends; curBackend++)
+	for (curBackend = 0; curBackend < TotalProcs; curBackend++)
```

Comment:
I'm not entirely against it, but I don't really see a point of continuing beyond `MaxBackends`

Reply:
True, I also noted that in the issue description. I'll update on some places where we're definitely looking for distributed transactions such as here.

Also added a comment for describing the distinction between usage of `MaxBackends` vs `TotalProcs`